src/libbson/src/bson/bson-atomic.h

```diff
@@ -67,6 +67,13 @@ enum bson_memory_order {
 #define BSON_IF_GNU_LEGACY_ATOMICS(...)
 #endif
 
+/* CDRIVER-4229 zSeries with gcc 4.8.4 produces illegal instructions for int and
+ * int32 atomic intrinsics. */
+#if defined(__s390__) || defined(__s390x__) || defined(__zarch__)
```

Comment:
These macros were determined from https://sourceforge.net/p/predef/wiki/Architectures/. `gcc` on the `zseries-rhel72` distro defines all three.